tests/kernel/fatal/exception/src/main.c

```diff
@@ -90,6 +90,38 @@ void entry_cpu_exception(void *p1, void *p2, void *p3)
 	rv = TC_FAIL;
 }
 
+void entry_cpu_exception_extend(void *p1, void *p2, void *p3)
+{
+	expected_reason = K_ERR_CPU_EXCEPTION;
+
+#if defined(CONFIG_ARM64)
+	__asm__ volatile ("svc 0");
+#elif defined(CONFIG_CPU_CORTEX_R)
+	__asm__ volatile ("BKPT");
+#elif defined(CONFIG_ARM)
```

Comment:
Could this be CPU_CORTEX_M, to make it explicit? That's because CONFIG_ARM is common to AARCH32 -R and -M

Reply:
Perhaps you can also trigger the SVC with the SVC instruction directly?

Reply:
> Perhaps you can also trigger the SVC with the SVC instruction directly?

Thank you for you advice. I tried to use SVC directly. But it failed on build qemu_cortex_a53,
/tmp/ccWs7opR.s:1185: Error: missing immediate expression at operand 1 -- `svc'

Reply:
> Could this be CPU_CORTEX_M, to make it explicit? That's because CONFIG_ARM is common to AARCH32 -R and -M

I distinguish CONFIG_CPU_CORTEX_R because the platform qemu_cortex_r5 will fail when using "swi 0", so I use another instruction to make fatal error insted of "swi 0".
When using CPU_CORTEX_M instead of CONFIG_ARM, building platform on mps2_an385 and mps2_521 will fail. The macro in  .config file show there is only CONFIG_ARM without CPU_CORTEX_M. So I think CONFIG_ARM is a more common macro to make the compiler build pass.